### PR TITLE
Support various java.time classes a JAX-RS Resource method params

### LINKED ISSUE #21663: Add support for more java.time as queryparams in resteasy-reactive
### Description

#15451 added support for java.time.LocalDate as a query param.

I need support for other java.time.* types, e.g.:
* LocalTime
* LocalDateTime

Ideally, Resteasy Reactive would support every java.time type out of the box.

### Implementation ideas

_No response_

### PR DESCRIPTION
**Pull request by geoand**
Furthermore, introduce the `@DateFormat` annotation that allows users
to customize the format of the incoming date value

Resolves: #21663

### CODE REVIEW

File: independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/parameters/converters/TemporalParamConverter.java [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+        //        if (parameter instanceof ZonedDateTime) {
+        //            if (formatter != null) {
+        //                return ZonedDateTime.parse(strValue, formatter);
+        //            } else {
+        //                return ZonedDateTime.parse(strValue);
+        //            }
+        //        }
+        //
+        //        throw new IllegalStateException(
+        //                "Type '" + parameter.getClass().getName() + "' is not supported as a JAX-RS Resource method parameter");
```
Comment (gsmet) [inline independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/parameters/converters/TemporalParamConverter.java:69]: Do we want to keep all this?

Comment (geoand) [reply independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/parameters/converters/TemporalParamConverter.java]: Good catch - removed

**Review by stuartwdouglas — APPROVED** [review]

**Comment by nilyin** [comment]
Hi, why is java.time.Instant support still missed?

**Comment by geoand** [comment]
Good point. https://github.com/quarkusio/quarkus/pull/24981 takes care of it

**Pull request merged**